### PR TITLE
Allowed Sites API ECS Cluster and Infrastructure

### PR DESCRIPTION
**Pull request by emileswarts**
Create a new ECS backend cluster for the Ruby backend

### CODE REVIEW

File: govwifi-backend-ruby/modules/ecs-autoscaling/launch-configuration.tf [hunk 1]
```diff
@@ -0,0 +1,225 @@
+resource "aws_launch_configuration" "ecs" {
+  name_prefix          = "tf-${var.Env-Name}-ruby-lc-"
+  image_id             = "${var.ami}"
+  instance_type        = "${var.instance_type}"
+  iam_instance_profile = "${var.instance-profile-id}"
+  key_name             = "${var.key_name}"
+  security_groups      = ["${split(",", var.security_group_ids)}"]
+
+  user_data = <<DATA
```
Comment (adrianclay) [inline govwifi-backend-ruby/modules/ecs-autoscaling/launch-configuration.tf]: Is this a dupe of the user_data from the other launch-config?

Be good to remove that duplication.

File: govwifi-backend-ruby/cluster.tf [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      "disableNetworking": null,
+      "dnsSearchDomains": null,
+      "portMappings": [
+        {
+          "hostPort": 80,
+          "containerPort": 80,
+          "protocol": "tcp"
+        },
+        {
+          "hostPort": 8080,
```
Comment (adrianclay) [inline govwifi-backend-ruby/cluster.tf]: 8080 ?

File: govwifi-backend-ruby/modules/ecs-autoscaling/launch-configuration.tf (hunk 1 above)
Comment (emileswarts) [reply govwifi-backend-ruby/modules/ecs-autoscaling/launch-configuration.tf]: We tried to solve this a while ago, by extracting it into a reusable file we could read.  
It was more complex than we anticipated.  
Propose we allow this to exist for now and add a ticket to come back to it.

Introducing Ansible for provisioning would also address this.

File: govwifi-allowed-sites-api/loadbalancer.tf [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    lb_port           = 8080
+    lb_protocol       = "http"
+  }
+
+  listener {
+    instance_port      = 80
+    instance_protocol  = "http"
+    lb_port            = 443
+    lb_protocol        = "https"
+    ssl_certificate_id = "${var.elb-ssl-cert-arn}"
```
Comment (afoldesi-gds) [inline govwifi-allowed-sites-api/loadbalancer.tf]: Are these re-generated for allowed-sites-api-elb....gov.uk?

File: govwifi-allowed-sites-api/loadbalancer.tf [hunk 4]
```diff
@@ -0,0 +1,53 @@
+resource "aws_elb" "allowed-sites-api-elb" {
+  count           = "${var.backend-elb-count}"
+  name            = "allowed-sites-api-elb-${var.Env-Name}"
+  subnets         = ["${var.subnet-ids}"]
+  security_groups = ["${var.elb-sg-list}"]
+
+  listener {
```
Comment (afoldesi-gds) [inline govwifi-allowed-sites-api/loadbalancer.tf]: Could probably get rid of the HTTP listeners here if we won't use them (the HTTPS listener rules below will "translate" requests to HTTP on the instance)

File: govwifi-allowed-sites-api/loadbalancer.tf (hunk 3 above)
Comment (emileswarts) [reply govwifi-allowed-sites-api/loadbalancer.tf]: Yeah just generated one before running the terraform

**Pull request merged**